src/main/scala/com/github/unchama/seichiassist/subsystems/gacha/bukkit/actions/BukkitDrawGacha.scala

```diff
@@ -0,0 +1,113 @@
+package com.github.unchama.seichiassist.subsystems.gacha.bukkit.actions
+
+import cats.effect.{ConcurrentEffect, IO, Sync, SyncIO}
+import com.github.unchama.minecraft.actions.OnMinecraftServerThread
+import com.github.unchama.seichiassist.concurrent.PluginExecutionContexts.onMainThread
+import com.github.unchama.seichiassist.subsystems.gacha.application.actions.{
+  DrawGacha,
+  LotteryOfGachaItems
+}
+import com.github.unchama.seichiassist.subsystems.gacha.domain.GachaPrizesDataOperations
+import com.github.unchama.seichiassist.subsystems.gacha.domain.GachaRarity.GachaRarity
+import com.github.unchama.seichiassist.subsystems.gacha.domain.GachaRarity.GachaRarity._
+import com.github.unchama.seichiassist.util.SendMessageEffect.sendMessageToEveryone
+import com.github.unchama.seichiassist.util._
+import net.md_5.bungee.api.chat.{HoverEvent, TextComponent}
+import org.bukkit.ChatColor._
+import org.bukkit.Sound
+import org.bukkit.entity.Player
+
+object BukkitDrawGacha {
+
+  import cats.effect.ConcurrentEffect.ops._
+
+  import scala.jdk.CollectionConverters._
+
+  import PlayerSendable._
+
+  def apply[F[_]: Sync: ConcurrentEffect: OnMinecraftServerThread](
+    implicit gachaPrizesDataOperations: GachaPrizesDataOperations[F]
+  ): DrawGacha[F, Player] = (player: Player, amount: Int) => {
+    OnMinecraftServerThread[F].runAction(SyncIO[Unit] {
+      val gachaLotteryItems = LotteryOfGachaItems.using.lottery(amount).toIO.unsafeRunSync()
+
+      val rarities: Vector[GachaRarity] = gachaLotteryItems.map { gachaPrize =>
+        if (gachaPrize.probability.value < 0.001) gigantic
+        else if (gachaPrize.probability.value < 0.01) big
+        else if (gachaPrize.probability.value < 0.1) regular
+        else potato
+      }
+      rarities.zip(gachaLotteryItems).map {
+        case (rarity, gachaPrize) =>
+          val givenItem = gachaPrize.getGiveItemStack(Some(player.getName))
+
+          /**
```

Comment:
```suggestion
          /*
```

`/**`で始まるブロックコメントはscaladoc用です